### PR TITLE
add guardian account management to student form

### LINKED ISSUE #493: better guardians accounts check
### what?

one of the user support flows we have consists on checking which email guardians should use to access their students accounts. the only way to do this, currently, is via the admin area, checking on the profiles page (which lists ALL the profiles in the DB)

### so what?

the admin area access is restricted to very few users (for a good reason). with the expansion of the support team, this becomes a problem, as this is a key information for offering a good user support experience

### now what?

users with "school_management" permission already have student and staff account management capabilities: the idea is to expand the scope and allow them to manage guardian accounts as well

### PR DESCRIPTION
**Pull request by PitzTech**
Issue: https://github.com/camino-school/lanttern/issues/493

Implementation: https://jam.dev/c/5d30105f-a12a-4bb7-9cd8-ef1d5ffcb9f5
This pull request introduces a robust system for managing guardian user accounts associated with students, primarily by allowing admins to add, update, and remove guardian emails through the student form. It adds new backend APIs for querying and updating guardian user accounts, enhances the student form UI to support multiple guardian emails, and ensures all operations are permission-checked and transactional. Extensive tests are included to guarantee correctness and edge-case handling.

**Guardian user account management:**

* Added `list_student_guardian_user_emails/2` and `set_student_guardian_user_accounts/3` functions in `Lanttern.Identity` to list and atomically update guardian user emails for a student, with strict permission and school checks. These functions ensure that for each email, a user and guardian profile are created if needed, and obsolete profiles are removed.
* Implemented helper functions in `Lanttern.Identity.User` for admin creation of users with random passwords and email validation, supporting automated guardian account creation.

**Student form UI/UX improvements:**

* Updated `StudentFormOverlayComponent` to display, add, and remove multiple guardian emails with live validation and error handling. Only users with `school_management` permission can manage guardian emails. [[1]](diffhunk://#diff-ca4e493f5c1eff8512aa8c0f537e3ce8fd173ca521e3c48846fecc0cfd7ce8e5L121-R179) [[2]](diffhunk://#diff-ca4e493f5c1eff8512aa8c0f537e3ce8fd173ca521e3c48846fecc0cfd7ce8e5R355-R381) [[3]](diffhunk://#diff-ca4e493f5c1eff8512aa8c0f537e3ce8fd173ca521e3c48846fecc0cfd7ce8e5L331-R435)
* Integrated backend calls so that saving a student also updates guardian user accounts transactionally, with clear error feedback if any operation fails.

**Testing and reliability:**

* Added comprehensive tests for the new guardian management APIs, covering creation, update, deletion, idempotency, permission checks, and edge cases such as whitespace emails and cross-school access.

**Code organization and support:**

* Refactored and extended imports and aliases where necessary to support the new features and maintain code clarity. [[1]](diffhunk://#diff-1df1268795274d57270f592994cfb5cdd52ab0ffe738bab788a9bc6eb26f0f80R13) [[2]](diffhunk://#diff-ca4e493f5c1eff8512aa8c0f537e3ce8fd173ca521e3c48846fecc0cfd7ce8e5R23-R25) [[3]](diffhunk://#diff-fb012cbd85dbb12b39fc4fbf4ccd0e4dcffb2edf9b7079774fc8be1321b0cb09R7)
* Ensured the form always displays at least one guardian email input for usability and consistent UX. [[1]](diffhunk://#diff-ca4e493f5c1eff8512aa8c0f537e3ce8fd173ca521e3c48846fecc0cfd7ce8e5R274-R275) [[2]](diffhunk://#diff-ca4e493f5c1eff8512aa8c0f537e3ce8fd173ca521e3c48846fecc0cfd7ce8e5L394-R569)

These changes collectively enable secure, flexible, and user-friendly management of guardian user accounts for students, with strong backend guarantees and a modernized frontend experience.

### CODE REVIEW

File: lib/lanttern_web/live/shared/schools/student_form_overlay_component.ex [hunk 1]
```diff
@@ -391,57 +492,72 @@ defmodule LantternWeb.Schools.StudentFormOverlayComponent do
...
+        student_or_nil,
+        student_params,
+        guardian_changes(socket),
+        guardian_emails
+      )
+
+    handle_save_result(result, action, socket)
+  end
+
+  defp handle_save_result(result, action, socket) do
```
Comment (endoooo) [inline lib/lanttern_web/live/shared/schools/student_form_overlay_component.ex]: when we have this pattern of a function where the only thing inside it is a case clause, we can implement this with multiple function clauses:

```elixir
defp handle_save_result({:ok, student}, action, socket) do
# ...
end

defp handle_save_result({:error, {:student, %Ecto.Changeset{} = changeset}}, action, socket) do
# ...
end

defp handle_save_result({:error, :guardian_accounts}, action, socket) do
# ...
end
```

**Review by endoooo — CHANGES_REQUESTED** [review]
## overview

functionality is 100%, and most pre review self-checks are ticked this time (I think you just missed translations). I'll ask just small refactors, detailed below.

## adjustments 

- [ ] refactor `student_form_overlay_component_test.exs` using `phoenix_test`
- [ ] missing translations
- [ ] use function clauses instead of `case` in `handle_save_result/3`

---

### missing translations

(always remember to run `gettext` mix tasks before opening the PR)

Image

**Review by endoooo — CHANGES_REQUESTED** [review]

**Review by endoooo — APPROVED** [review]
